### PR TITLE
testing

### PR DESCRIPTION
**Pull request by toniSta**

### CODE REVIEW

**Comment by toniSta** [comment]
Fixes all tests. Please check, if docker setup still works, since I rewrote `config.py`

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Blockchain persistency related files
 blockchain/blockchain_files/*
+tests/blockchain_files/*
```
Comment (Benedikt1992) [inline .gitignore]: Please move this line one line down

File: blockchain/helper/block_validator.py [hunk 2]
```diff
@@ -58,21 +60,22 @@ def validate_block(block, previous_block):
...
-             logger.info("Block contains invalid transactions")
-             return False
+            logger.info("Block contains invalid transactions")
+            return False
 
     # Block has no transactions
-    # if len(block.transactions) == 0:
-    #     logger.info("Block does not contain any transaction")
-    #     return False
+    if len(block.transactions) == 0:
```
Comment (Benedikt1992) [inline blockchain/helper/block_validator.py]: Don't enable this check

File: blockchain/helper/block_validator.py [hunk 3]
```diff
@@ -58,21 +60,22 @@ def validate_block(block, previous_block):
         return False
 
     # Check if all transactions are valid
-    admissions, doctors, vaccines = Chain().get_registration_caches_by_blockhash(block.previous_block)
+    admissions, doctors, vaccines = Chain().\
+        get_registration_caches_by_blockhash(block.previous_block)
     for transaction in block.transactions:
         if not transaction.validate(admissions, doctors, vaccines):
-             logger.info("Block contains invalid transactions")
```
Comment (Benedikt1992) [inline blockchain/helper/block_validator.py]: Maybe this could be the cause for the error

File: tests/test_block_validation.py [hunk 4]
```diff
@@ -1,111 +1,130 @@
...
+PUBLIC_KEY = None
+PRIVATE_KEY = None
 
-PUBLIC_KEY = RSA.import_key(open("tests" + os.sep + "testkey_pub.bin", "rb").read())
-PRIVATE_KEY = RSA.import_key(open("tests" + os.sep + "testkey_priv.bin", "rb").read())
 
+def setup_module(module):
+    # blockchain files folder will be deleted by create_initial_block
+    global GENESIS, PUBLIC_KEY, PRIVATE_KEY
+    GENESIS = create_initial_block()
```
Comment (Benedikt1992) [inline tests/test_block_validation.py]: Does this overwrite the prod genesis block? (Why not?)

Comment (toniSta) [reply tests/test_block_validation.py]: It doesn't overwrite the prod block. In the config file, I detect, whether the script was called by pytest (which is actually not a nice way, but I had no better idea). Based on that, I set the blockchain files folder to `tests/blockchain_files` or  `blockchain/blockchain_files`. When creating the genesis block, it retrieves the path from config.

**Comment by toniSta** [comment]
I accidentally added wrong public and private keys, which were created during testing. I checked out the correct ones and it seems to be working now. When running the tests, keys will be written into a separate folder. Hence, the error will (hopefully) not happen again. All errors should be fixed now. Feel free to test it once again. :smile:

**Comment by alpreu** [comment]
Tested and looks good

**Pull request merged**